chrome/content/zotero/xpcom/data/library.js

```diff
@@ -567,6 +570,22 @@ Zotero.Library.prototype.updateSearches = Zotero.Promise.coroutine(function* ()
 	this._hasSearches = !!(yield Zotero.DB.valueQueryAsync(sql, this.libraryID));
 });
 
+Zotero.Library.prototype.hasItems = Zotero.Promise.coroutine(function* () {
+	if (! this.id) {
+		throw new Error("Library is not saved yet");
+	}
+	if (this._hasItems === null) {
+		yield this.updateItems();
+	}
```

Comment:
This will never be run, because `hasItems` is populated during library load. We can have a per-library hasItems — that's not technically the test we want for this, but I think it wouldn't matter because you can't have groups without a userID, and not having a userID is a prereq for this to be called — but it should still be run on-demand and not cached, since we're not updating it like the other two.

Right now, if you delete the DB, sync, get the warning, cancel, create a new item (or import thousands of items), and sync again, you get the warning again.